bin/behat-prepare.sh

```diff
@@ -22,125 +23,191 @@ fi
-openssl req -newkey rsa:2048 -new -x509 -days 3652 -nodes -out $PREPARE_DIR/private/simplesamlphp/cert/saml.crt -keyout $PREPARE_DIR/private/simplesamlphp/cert/saml.pem -batch
+openssl req -newkey rsa:2048 -new -x509 -days 3652 -nodes -out "$PREPARE_DIR"/private/simplesamlphp/cert/saml.crt -keyout "$PREPARE_DIR"/private/simplesamlphp/cert/saml.pem -batch
 
+TWIG_TEMPLATE_PATH="$PREPARE_DIR/private/simplesamlphp/modules/core/templates/loginuserpass.twig"
 # Modify the login template so Behat can submit the form
-sed -i  -- "s/<button/<button id='submit'/g" $PREPARE_DIR/private/simplesamlphp/modules/core/templates/loginuserpass.tpl.php
-sed -i  -- "s/this.disabled=true; this.form.submit(); return true;//g" $PREPARE_DIR/private/simplesamlphp/modules/core/templates/loginuserpass.tpl.php
-# Second button instance shouldn't have an id
-sed -i  -- "s/<button id='submit' class=\"btn\" tabindex=\"6\"/<button class=\"btn\" tabindex=\"6\"/g" $PREPARE_DIR/private/simplesamlphp/modules/core/templates/loginuserpass.tpl.php
-
-cd $PREPARE_DIR
-# Make the SimpleSAMLphp installation publicly accessible
-ln -s ./private/simplesamlphp/www ./simplesaml
+echo "Operating on: $TWIG_TEMPLATE_PATH"
+sed -i  -- "s/<button class=\"pure-button pure-button-red pure-input-1-2 pure-input-sm-1-1 right\" id=\"submit_button\"/<button class=\"pure-button pure-button-red pure-input-1-2 pure-input-sm-1-1 right\" id=\"submit\"/g" "$TWIG_TEMPLATE_PATH"
+sed -i  -- "s/Login/Submit/g" "$TWIG_TEMPLATE_PATH"
+# Modify the loginuserpass.js file so Behat can submit the form
+JS_FILE_PATH="$PREPARE_DIR/private/simplesamlphp/modules/core/public/assets/js/loginuserpass.js" # Adjust path if necessary
+sed -i -- 's/getElementById("submit_button")/getElementById("submit")/g' "$JS_FILE_PATH"
+sed -i -- 's/button.disabled = true;//g' "$JS_FILE_PATH"
+
+composer install --no-dev --working-dir="$PREPARE_DIR"/private/simplesamlphp --ignore-platform-req=ext-ldap
+
+# Copy SimpleSAMLphp installation into public /simplesaml directory.
+cd "$PREPARE_DIR"
+mkdir -p "$PREPARE_DIR"/simplesaml
+cp -r "$PREPARE_DIR"/private/simplesamlphp/public/* "$PREPARE_DIR"/simplesaml
+cp -r "$PREPARE_DIR"/private/simplesamlphp/vendor "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/src "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/modules "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/config "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/templates "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/cert "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/metadata "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/routing "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/attributemap "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/lib "$PREPARE_DIR"/simplesaml/
```

Comment:
Why are these copied, don't we end up with two versions in vc (yeah yeah it's thrown away I know). Why not ln or mv?

Reply:
well it is helpful to have the full checkout of simplesamlphp vs the public version (with an altered directory structure) to compare back to. we can't link because the structure doesn't match 100% (I mean we could ln all of these lines) and moving would leave pieces of the original checkout behind because we don't copy everything. 

having both was helpful when i could compare the original directory structure especially considering that. we modify that for the tests.